.github/workflows/yaml-lint-checker.yaml

```diff
@@ -0,0 +1,68 @@
+---
+name: Yaml Lint Release Notes Pull Request
+on:
+  pull_request:
+    paths:
+      - releases/**/release-notes/**.yaml
+  # Allow manual triggering
+  workflow_dispatch: {}
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+jobs:
+  yaml-lint-checker:
+    name: Release Notes Check Yaml 👀
+    runs-on: ubuntu-22.04
+    if: ${{ !github.event.pull_request.draft }}
+    steps:
+      - name: Cancel Previous Actions
+        uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: '0'
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22.4' 
+      - name: Install go dependencies
+        run: go mod download
+      - name: Build go prog
+        run: go build -o validate_yaml .github/scripts/validate_yaml.go
+      - name: Validate YAML
+        id: validate_yaml
+        run: |
+          INVALID_FILES=""
+          # check diff for invalid yaml files
+          git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.sha }} -- releases/ | grep -E '\.ya?ml$'
+          CHANGED_FILES=$(git diff --name-only $(git merge-base origin/master HEAD) -- releases/ | grep -E '\.ya?ml$' || true)
```

Comment:
for this part you maybe can use this action that get all you need: https://github.com/tj-actions/changed-files

Reply:
checking this